webapp/src/components/ClassOverlapTable.test.tsx

```diff
@@ -0,0 +1,77 @@
+import { QueryPipelineState } from "types/models";
+
+const renderClassOverlapTable = (
+  availableDatasetSplits: AvailableDatasetSplits,
+  pipeline: QueryPipelineState
+) =>
+  renderWithRouterAndRedux(
+    <ClassOverlapTable
+      jobId="local"
+      pipeline={pipeline}
+      availableDatasetSplits={availableDatasetSplits}
+    />,
+    { route: "/local?pipeline_index=0", path: "/:jobId?:pipeline" }
+  );
+
+describe("ClassOverlapTable", () => {
+  const handlers = [getClassOverlapAPIResponse];
+  const server = setupServer(...handlers);
+
+  beforeAll(() => server.listen());
+  afterEach(() => server.resetHandlers());
+  afterAll(() => server.close());
+
+  it("should have expected columns", () => {
+    renderClassOverlapTable({ train: true, eval: true }, { pipelineIndex: 0 });
+    const expectedColumnHeaders = [
+      "Source Class",
+      "Target Class",
+      "Semantic Overlap Score",
+      "Pipeline Confusion",
+      "Utterances with Overlap",
+    ];
+    const columnHeaderDescription = [
+      "The class label of the samples being analyzed.",
+      "The class that the source class may look like. For pipeline confusion, this is the prediction.",
+      "Class overlap measures the extent to which source class samples are semantically similar to target class samples, in the training data. The score comes from both the proportion of samples and their degree of similarity.",
+      "Pipeline confusion indicates whether source class samples in the evaluation set are predicted to be in the target class.",
+      "Percent of source class samples that semantically overlap the target class (all in the training set).",
+    ];
+    const actualColumnHeaders = screen.getAllByRole("columnheader");
+    expectedColumnHeaders.forEach((name, index) => {
+      expect(actualColumnHeaders[index].textContent).toEqual(name);
+      within(actualColumnHeaders[index]).getByLabelText(
+        columnHeaderDescription[index]
+      );
+    });
+  });
+
+  it("should display the default sort as desc for colum header 'Semantic Overlap Score'", () => {
+    renderClassOverlapTable({ train: true, eval: true }, { pipelineIndex: 0 });
+    const header = screen.getByRole("columnheader", {
+      name: /Class overlap/,
+    });
+    expect(header).toHaveAttribute("aria-sort", "descending");
+  });
+
+  it("should not display the columns except 'Pipeline Confusion' if no pipeline exist/selected", () => {
+    renderClassOverlapTable(
+      { train: true, eval: true },
+      { pipelineIndex: undefined }
+    );
+    expect(
+      screen.queryByRole("columnheader", { name: /Pipeline confusion/ })
+    ).toBeNull();
+  });
+
+  it("should not display the Footer component if the number of rows is lesser than initial number", async () => {
+    renderClassOverlapTable({ train: true, eval: true }, { pipelineIndex: 0 });
+    await waitFor(() => expect(screen.queryByText(/See more/)).toBeNull());
+  });
```

Comment:
The `waitFor()` doesn't have any effect here, as `screen.queryByText(/See more/)` will be null from the start. So, no need for the `async`, and also there is a typo:
```suggestion
  it("should not display the Footer component if the number of rows is less than initial number", () => {
    renderClassOverlapTable({ train: true, eval: true }, { pipelineIndex: 0 });
    expect(screen.queryByText(/See more/)).toBeNull();
  });
```

Reply:
Okay, I suggest using async, as we display the footer component based on the length of the row. Hence, we should wait for the API response to be fetched. The purpose of this test case is to check the display of the "see more" section after the API response. Ideally, it should be null because my mock API response has only 3 three records.

Reply:
But here, you do NOT "wait for the API response to be fetched". I guess you should. But right now, the test `toBeNull()` will pass on the very first loop of `waitFor()`, and it won't wait for the API response to be fetched.

Reply:
Will you try to wait for the API response to be fetched? Like adding an `expect()` for something that shows that the API results are loaded, and then checking for the `See more` button?

Reply:
Right. Good approach 👍 . I added a test scenario to check "MuiDataGrid-row" is loaded to ensure API has successfully fetched the records then followed by testing the footer component.